internal/resource-request-operator/offerUpdater.go

```diff
@@ -0,0 +1,114 @@
+package resourcerequestoperator
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	"github.com/liqotech/liqo/internal/resource-request-operator/interfaces"
+	"github.com/liqotech/liqo/pkg/discovery"
+)
+
+const requeueTimeout = 5 * time.Second
```

Comment:
Unclear to me why you are combining a tight requeue timeout with a push mechanism. I feel either one of the two approaches should be sufficient (if you use a push mechanism, then a requeue of half the TTL plus a random delta for de-sync should be sufficient, otherwise you drop the push mechanism and keep only the requeue)